lib/live_debugger/gen_servers/ets_table_server.ex

```diff
@@ -97,6 +104,28 @@ defmodule LiveDebugger.GenServers.EtsTableServer do
     {:noreply, updated_state}
   end
 
+  @impl true
+  def handle_call(:garbage_collect, _from, state) do
+    state
+    |> Enum.map(fn {_, %TableInfo{table: table} = table_info} ->
+      {table_info, Memory.table_size(table)}
+    end)
+    |> Enum.each(fn {%TableInfo{table: table, watchers: watchers}, size} ->
+      cond do
+        Enum.empty?(watchers) and size > max_table_size(:non_watched) ->
+          trim_ets_table(table, max_table_size(:non_watched))
+
+        size > max_table_size(:watched) ->
+          trim_ets_table(table, max_table_size(:watched))
```

Comment:
I think it could be a bit easier to read it you created private functions with decent names explaining what these conditions do